src/v1/KaitaiServices.ts

```diff
@@ -72,14 +72,22 @@ export class CompilerService {
     ksySchema: KsySchema.IKsyFile;
     ksyTypes: IKsyTypes;
 
-    compile(srcYamlFsItem: IFsItem, srcYaml: string, kslang: string, debug: true | false | "both"): Promise<any> {
+    async compile(srcYamlFsItem: IFsItem, srcYaml: string, kslang: string, debug: true | false | "both"): Promise<any> {
         var perfYamlParse = performanceHelper.measureAction("YAML parsing");
 
         this.jsImporter.rootFsItem = srcYamlFsItem;
 
         try {
             this.ksySchema = <KsySchema.IKsyFile>YAML.parse(srcYaml);
-            this.ksyTypes = SchemaUtils.collectKsyTypes(this.ksySchema);
+            if (this.ksySchema.meta.imports) {
+                let imports = this.ksySchema.meta.imports.map(item => this.jsImporter.importYaml(item, "rel"))
```

Comment:
@VitaliyZaharenko Well, my attention is apparently getting worse and I didn't notice several problems:

1. Your code has no semicolons, unlike the rest of the codebase.

2. The highlighted line doesn't account for absolute imports (i.e. loading from the preloaded `kaitai-io/formats` library). That's very bad, because loading a `.ksy` spec with absolute imports from the formats library (for example [`standard_midi_file`](https://github.com/kaitai-io/kaitai_struct_formats/blob/7f3ecee038ae4aa14da7d9a80d8abfd4080c9b59/media/standard_midi_file.ksy#L17-L18)) will trigger an error `Parse error: undefined\nCall stack: undefined`, because a relative path `/common/vlq_base128_be` cannot be found (it ends up looking for https://ide.kaitai.io/devel/formats/media//common/vlq_base128_be.ksy, which of course doesn't exist).
   
   It is so bad that I reverted this PR for now (https://github.com/kaitai-io/kaitai_struct_webide/commit/73104cce5acf819221d7cf41d6e28df4d8be0b51) - feel free to open a new PR with the problems fixed.

Reply:
@generalmimon 
Thanks for the feedback, I will correct the issues as soon as I have a little more free time.
I also noticed that the problem of transitive imports remains. info is currently loaded only for direct dependencies.